freecad/ship/ShipGui.py

```diff
@@ -232,6 +232,43 @@ def GetResources(self):
                 'MenuText': MenuText,
                 'ToolTip': ToolTip}
 
+class Amadeo:
+    def IsActive(self):
```

Comment:
Apparently the tools can be used without any actual ship, right? So you can just write the data manually.

Then I suppose you want to replace that by a `return True`

Reply:
I used the example ship to get the main dimensions. However, I think it was a good idea to be able to use the prediction without the ship. Apart from this file, I have modified the TaskPanel and Plotaux files, so that I can get the resistance without having a vessel selected. In addition, I have introduced in the UI the main dimensions and coded in TaskPanel to fill those values whenever it is possible because there is a ship selected and if not, return 0. I have tested that all the changes introduced work.

Reply:
Nice!